src/js/homepage.js

```diff
@@ -2,6 +2,78 @@ import React from "react";
 import ReactDOM from "react-dom";
 import _ from "lodash";
 
+class TwitterTabs extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      activeTab: '',
+    }
+  }
+
+  openTab(tabId) {
+    let tabContent = document.getElementsByClassName('tabContent');
+    _.forEach(tabContent, content => {
+      content.style.display = 'none';
+    });
+
+    this.setState({
+      activeTab: tabId,
+    });
+
+    document.getElementById(tabId).style.display = 'block';
+  }
+
+  render() {
+    return React.createElement('ul', {className: 'tabUL'},
+        React.createElement('li', {type: 'none', className: 'tab' + (this.state.activeTab === 'timelineColumn' ? ' active' : ''), id: 'defaultTab', onClick: () => this.openTab('timelineColumn')}, 'Home Timeline'),
+        React.createElement('li', {type: 'none', className: 'tab' + (this.state.activeTab === 'userTweetsColumn' ? ' active' : ''), onClick: () => this.openTab('userTweetsColumn')}, 'User Tweets'),
+        React.createElement('li', {type: 'none', className: 'tab' + (this.state.activeTab === 'postTweetColumn' ? ' active' : ''), onClick: () => this.openTab('postTweetColumn')}, 'Post Tweet'));
+  }
+}
+
+class PostTweetUI extends React.Component {
+  constructor(props){
+    super(props);
+    this.state = {
+      numChars: 0,
+      postTweetButtonDisabled: true,
+      userTweetText: '',
+      postStatus: 'pending',
+    }
+  }
+
+  preparePost() {
+    let userTweetText = document.getElementById('userTweetText').value;
```

Comment:
since preparePost is executed by a callback, you could pull the object through a param
i.e.
preparePost(event) {
 let userTweet = event.target.value;